ocaml/xapi/vm_sysprep.ml

```diff
@@ -275,19 +280,8 @@ let sysprep ~__context ~vm ~unattend ~timeout =
   call ~__context @@ fun rpc session_id ->
   Client.VBD.insert ~rpc ~session_id ~vdi ~vbd ;
   Thread.delay !Xapi_globs.vm_sysprep_wait ;
-  match trigger ~domid ~uuid ~timeout with
-  | true -> (
-      debug "%s: sysprep running, ejecting CD" __FUNCTION__ ;
-      try eject ~rpc ~session_id iso vbd
-      with exn ->
-        debug "%s: ejecting CD failed: %s" __FUNCTION__ (Printexc.to_string exn) ;
-        fail VM_CDR_eject
-    )
+  match trigger ~rpc ~session_id ~domid ~uuid ~timeout ~vbd ~iso with
+  | true ->
+      ()
   | false ->
-      debug "%s: sysprep timeout, ejecting CD" __FUNCTION__ ;
-      ( try eject ~rpc ~session_id iso vbd
```

Comment:
We no longer have this, what if any exception is raised from sysprep, or sysprep itself times out? Shouldn't we try to eject the CD?

Reply:
You are right; if sysprep does not start or finish within its respective timeout, we would not eject the CD. I'll try to use a finally to ensure this. It feels like this is increasingly delicate.

Reply:
We could (Sys.remove, which I assume is unlink) the file itself, although that'll then result in the VM failing to boot due to a missing VDI.